javascript/javascript1/week4/homework_exercises/voice_assistant.js

```diff
@@ -0,0 +1,164 @@
+const userInfo = {
+    name: undefined,
+    todo: [],
+}; 
+
+function getReply (command) {
+    const lowCaseCommand = command.toLowerCase();
+
+    // greeting
+    if (lowCaseCommand.includes("my name is")) {
+        const firstName = command.split("is ")[1];
+        const reply = (userInfo.name === firstName) ? `Hi again, ${firstName}!` : `Nice to meet you ${firstName}!`;
+
+        userInfo.name = firstName;
+        return reply;
+    }
+
+    // answering name
+    if (lowCaseCommand.includes("what") && lowCaseCommand.includes("my name")) {
+        const reply =  (userInfo.name !== undefined) ? `Your name is ${userInfo.name}.` : `I'm sorry, I don't know your name.`;
+        return reply;
+    }
+
+    // identify the task
+    function taskIdentifier (lowCaseCommand) {
+        const arraySentence = lowCaseCommand.split(" ");
+        const wordsToRemove = ["add", "to", "my", "todo", "list", "remove", "from", "do"]
+        const filteredWords = arraySentence.filter(word => !wordsToRemove.includes(word))
+        const task = filteredWords.join(" ")
+        return task;
+    }
+
+    // adding to todo list
+    if (lowCaseCommand.includes("add") && lowCaseCommand.includes("todo")) {
+        const task = taskIdentifier(lowCaseCommand);
+
+        const reply =  userInfo.todo.includes(task) ? `Couldn't add ${task} because it already exists in your todo list.` : `Successful addition of ${task} to your todo list.`;
+
+        if (!userInfo.todo.includes(task)) {
+            userInfo.todo.push(task);
+        }
+
+        return reply;
+    }
+
+    // remove from todolist
+    if ((lowCaseCommand.includes("remove") && lowCaseCommand.includes("todo"))) {
+        const task = taskIdentifier(lowCaseCommand);
+
+        const reply =  userInfo.todo.includes(task) ? `Successful removal of ${task} from your todo list.` : `Couldn't remove ${task} because it doesn't exist in your todo list yet.`;
+
+        if (userInfo.todo.includes(task)) {
+            const itemToRemoveIndex = userInfo.todo.indexOf(task);
+            userInfo.todo.splice(itemToRemoveIndex);
+        }
+
+        return reply;
+    }
+
+    // show todolist
+    if ((lowCaseCommand.includes("what") && lowCaseCommand.includes("in") && lowCaseCommand.includes("todo"))) {
+        return userInfo.todo;
+    }
+
+    // today's date
+    if (lowCaseCommand.includes("what day") && lowCaseCommand.includes("today")) {
+        const today = new Date()
+
+        const day = today.getDate();
+        const month = today.toLocaleString('default', {month: 'long'}); //this one was tough
+        const year = today.getFullYear();
+
+        return `Today is ${day} of ${month}, ${year}`;
+        
+    }
+
```

Comment:
Now you have a great toolbox, and as you progress, you'll discover some tools that replace others, part of the discovery journey ☯️